ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx

```diff
@@ -0,0 +1,338 @@
+          <form>
+            <RadioGroup value={mode} onChange={handleChange}>
+              <FormControlLabel
+                value="studio"
+                control={<Radio color="primary" />}
+                label={
+                  <ListItemText
+                    primary={
+                      <FormattedMessage
+                        id="publishOnDemand.pathModeDescription"
+                        defaultMessage="Publish changes made in Studio via the UI"
+                      />
+                    }
+                    secondary="By path"
+                  />
+                }
+                className={classes.byPathModeSelector}
+              />
+              <FormControlLabel
+                value="git"
+                control={<Radio color="primary" />}
+                label={
+                  <ListItemText
+                    primary={
+                      <FormattedMessage
+                        id="publishOnDemand.tagsModeDescription"
+                        defaultMessage="Publish changes made via direct git actions against the repository or pulled from a remote repository"
+                      />
+                    }
+                    secondary="By tags or commit ids"
+                  />
+                }
+              />
+            </RadioGroup>
+          </form>
+        </Paper>
+
+        <Collapse in={nnou(mode)} timeout={300} unmountOnExit>
+          <PublishOnDemandForm
+            formData={mode === 'studio' ? publishStudioFormData : publishGitFormData}
+            setFormData={mode === 'studio' ? setPublishStudioFormData : setPublishGitFormData}
+            mode={mode}
+            publishingTargets={publishingTargets}
+            publishingTargetsError={publishingTargetsError}
+          />
+
+          <div className={classes.noteContainer}>
+            <Typography variant="caption" className={classes.note}>
+              {mode === 'studio' ? formatMessage(messages.publishStudioNote) : formatMessage(messages.publishGitNote)}
+              <Link href="#" onClick={toggleMode} className={classes.noteLink}>
+                {mode === 'studio' ? formatMessage(messages.publishStudioLink) : formatMessage(messages.publishGitLink)}
+              </Link>
+            </Typography>
+          </div>
+        </Collapse>
+      </div>
+
+      <div className={classes.actions}>
+        <Button variant="outlined" color="default" onClick={onCancel} className={classes.cancelBtn} disabled={!mode}>
+          <FormattedMessage id="words.cancel" defaultMessage="Cancel" />
+        </Button>
+        <Button
+          variant="contained"
+          color="primary"
+          disabled={!(mode === 'studio' ? publishStudioFormValid : publishGitFormValid)}
+          onClick={mode === 'studio' ? bulkPublishConfirmation : publishBy}
+        >
+          <FormattedMessage id="words.publish" defaultMessage="Publish" />
+        </Button>
+      </div>
```

Comment:
- This whole chunk shouldn't show up when `!mode`.
- Would using DialogFooter remove the need of a custom class for this footer?

Reply:
it does, updating it to use DialogFooter